lib/diaspora/fetcher/public.rb

```diff
@@ -104,16 +104,13 @@ def process_posts
 
         logger.debug "post: #{post.to_s[0..250]}"
 
-        entry = StatusMessage.diaspora_initialize(
-          author:                @person,
-          public:                true,
-          guid:                  post["guid"],
-          text:                  post["text"],
-          provider_display_name: post["provider_display_name"],
-          created_at:            ActiveSupport::TimeZone.new("UTC").parse(post["created_at"]).to_datetime,
+        DiasporaFederation::Federation::Fetcher.fetch_public(
+          @person.diaspora_handle,
+          :post,
+          post["guid"]
         )
-        entry.save
-
+      rescue DiasporaFederation::Federation::Fetcher::NotFetchable => e
+        logger.debug e.message
```

Comment:
As this should basically never happen (as if the post was just there before, it should always be fetchable, unless it really gets deleted in exactly this moment), I'm not sure if it should be logged as warning (as in production we don't have debug logging enabled, and it might be interesting to see if it happens) ... on the other hand, if it basically should never happen, it might as well be fine if it just gets silently dropped in production? :man_shrugging:

Reply:
I think it might happen in case of some network instability, or for example if the pod is overloaded, or if there is a bug and for some reason fetching the post responds `500`. So I think it might be, though it's unlikely. So I agree it's worth making it more visible -- it shouldn't be very noisy but it could be interesting to see when this happens and how othen.